src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilterElement.java

```diff
@@ -237,27 +206,4 @@ private List<Item> getItems(TreeWalkerAuditEvent event) {
         }
         return items;
     }
-
-    @Override
-    public int hashCode() {
-        return Objects.hash(filePattern, checkPattern, messagePattern,
```

Comment:
please restore such methods and do `filePattern.patern(),.....` to simulate how it was before.